sdl.cpp

```diff
@@ -0,0 +1,294 @@
+#include <string>
+#include <SDL.h>
+#include <SDL_image.h>
+#include <SWI-cpp2.h>
+
+PREDICATE(sdl_init_, 1)
+{
+  if (SDL_Init(A1.as_uint32_t()) < 0)
+  {
+    throw PlException(PlTerm_string(SDL_GetError()));
+  }
+  return true;
+}
+
+PREDICATE(sdl_quit, 0)
+{
+  SDL_Quit();
+  return true;
+}
+
+PREDICATE(sdl_createwindow_, 7)
+{
+  SDL_Window *window = SDL_CreateWindow(A2.as_string().c_str(),
+                                        A3.as_int(), A4.as_int(),
+                                        A5.as_int(), A6.as_int(),
+                                        A7.as_uint32_t());
+  if (window == NULL)
+  {
+    throw PlException(PlTerm_string(SDL_GetError()));
+  }
+  return A1.unify_pointer(window);
```

Comment:
Using the "blob" interface provides a bit more functionality than using a pointer:
* you can store more info with the blob, which  might help with debugging (e.g., the parameters to SDL_CreateWindow)
* the blob can be atom garbage collected, which can do the close "automatically" for you (although it's probably better to do the close explicitly).

If you wish, I can probably put together a blob interface for you (I'd add another blob example to the documentation).

Reply:
So I already did a first attempt at using a blob interface but with the C interface.
I really liked the debugging aspect of the blob interface as well as the possibility to "type" the blob so that we can differentiate between a `window` blob or a `surface` blob.

If you could add in the documentation an example of creating a blob for an **opaque** pointer, I would use it :)
I'm saying **opaque** because the SDL do not expose the definition of the structure the pointer is pointing to.
For example, here, the `window` pointer points to a structure called `SDL_Window`, but the SDL header does not expose the definition of the `SDL_Window` structure.

However, I don't think I'm going to use the automatic freeing of blobs because I have noticed that it can sometimes cause segmentation crash when developping. (changing the code, doing `make` multiple times).
As discussed in the discourse thread, this will need a more complex solution.